### ISSUE TITLE
Ipython console mixes up Input/Output during KeyError

### ISSUE THREAD

**Issue by PabTorre** [spyder-ide/spyder#4878]
## Description of your problem

**What steps will reproduce the problem?**

1. Open Ipython
2. run: raise KeyError("This is a key error")

**What is the expected output? What do you see instead?**

Expected Output: 

In [1]: raise KeyError("This is a key error")

Traceback (most recent call last):

  File "<ipython-input-1-d924429c7516>", line 1, in <module>
    raise KeyError("This is a key error")

KeyError: 'This is a key error'
In[2]:

Current Output: 
In [1]: raise KeyError("This is a key error")

In[2]: Traceback (most recent call last):

  File "<ipython-input-1-d924429c7516>", line 1, in <module>
    raise KeyError("This is a key error")

KeyError: 'This is a key error'

**Please provide any additional information below**

The traceback for most recent call is being placed under the next In cell, instead of being treated as an output. 
As a result using the arrows, delete or enter has no effect on the console. 
But if one uses the mouse to put the pointer at the end of the text, and presses Enter; a SyntaxError happens as ipython tries to interpret the Traceback as if it was code. 

If instead, one presses CTRL+Backspace, the text from the Traceback is deleted and one can enter any code in the cell afterwards. 

## Version and main components

* Spyder Version: 3.1.4 
* Python Version: 3.6.1
* Qt Versions: 5.6.2, PyQt5 5.6 on Windows
* OS: Windows 7

## Dependencies
```
pyflakes >=0.6.0 :  1.5.0 (OK)
pep8 >=0.6       :  1.7.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
qtconsole >=4.2.0:  4.3.0 (OK)
nbconvert >=4.0  :  5.1.1 (OK)
pandas >=0.13.1  :  0.20.1 (OK)
numpy >=1.7      :  1.12.1 (OK)
sphinx >=0.6.6   :  1.5.4 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
jedi >=0.9.0     :  0.9.0 (OK)
matplotlib >=1.0 :  2.0.1 (OK)
sympy >=0.7.3    :  1.0 (OK)
pylint >=0.25    :  1.6.4 (OK)
```
## Screenshots 
Raise Keyerror:
Observe that the traceback output is set as input for cell [2]
![image](https://user-images.githubusercontent.com/4969082/28843995-33fa5622-76d1-11e7-9867-cd51c43509a8.png)

If one presses Enter after selecting the Traceback with a mouse click:

![image](https://user-images.githubusercontent.com/4969082/28844039-4a159cd2-76d1-11e7-9f72-dab24a03f0b4.png)

**Comment by ccordoba12** [comment]
Thanks for reporting. Although ugly, this error is not that common, so we don't plan to address it right now.

**Comment by ccordoba12** [comment]
This is a duplicate of issue #4504. We'll fix it in Spyder **3.2.2**.
